File: Easy/32SmallestDifference.py
def smallestDifference(arrayOne, arrayTwo):
    # Write your code here.
    arrayOne.sort()
    arrayTwo.sort()
    idxOne = 0
    idxTwo = 0
    smallestNumber = float('inf')
    currentNumber = float('inf')
    smallestPair = []
    while idxOne < len(arrayOne) and idxTwo < len(arrayTwo):
        firstNumber = arrayOne[idxOne]
        secondNumber = arrayTwo[idxTwo]
        if firstNumber < secondNumber:
            currentNumber = secondNumber - firstNumber
            idxOne += 1
        elif secondNumber < firstNumber:
            currentNumber = firstNumber - secondNumber
            idxTwo += 1
        else:
            return [firstNumber, secondNumber]

        if currentNumber < smallestNumber:
            smallestNumber = currentNumber
            smallestPair = [firstNumber, secondNumber]

    return smallestPair


arrayOne = [-1, 5, 10, 20, 28, 3]
arrayTwo = [26, 134, 135, 15, 17]
print(smallestDifference(arrayOne, arrayTwo))

File: Medium/32SmallestDifference.js
function smallestDifference(arrayOne, arrayTwo) {
  // Write your code here.
  arrayOne.sort((a, b) => a - b);
  arrayTwo.sort((a, b) => a - b);
  let smallestNumber = Infinity;
  let currentNumber = Infinity;
  let smallestPair = [];
  let idxOne = 0;
  let idxTwo = 0;
  while (idxOne < arrayOne.length && idxTwo < arrayTwo.length) {
    let firstNumber = arrayOne[idxOne];
    let secondNumber = arrayTwo[idxTwo];
    if (firstNumber < secondNumber) {
      currentNumber = secondNumber - firstNumber;
      idxOne++;
    } else if (secondNumber < firstNumber) {
      currentNumber = firstNumber - secondNumber;
      idxTwo++;
    } else {
      return [firstNumber, secondNumber];
    }

    if (currentNumber < smallestNumber) {
      smallestNumber = currentNumber;
      smallestPair = [firstNumber, secondNumber];
    }
  }
  return smallestPair;
}
let arrayOne = [-1, 5, 10, 20, 28, 3];
let arrayTwo = [26, 134, 135, 15, 17];
console.log(smallestDifference(arrayOne, arrayTwo));
